tools/customization_dataset_preparation/customization_dataset_preparation.py

```diff
@@ -79,31 +79,50 @@ def load_file_into_df(filename):
     return df, message
 
 
+def recommend_hyperparameters_human_readable(recommended_hyperparameters):
+    message = 'TODO: Recommended hyperparameters\n'
+    for param, param_value in recommended_hyperparameters.items():
+        message += f'{param}: {param_value}\n'
+    return message
+
+
 def recommend_hyperparameters(df, model=None):
     """
     Makes recommendations on the batch_size to use for training, based on the dataset size
+    
+    All hyperparameters except batch_size and max_batch_size are hardcoded based on API defaults for now
     """
     potential_batch_sizes = [2, 4, 8, 12, 16, 32, 64, 128]
     bs = 2
     for potential_bs in potential_batch_sizes:
         if 0.002 * len(df) > potential_bs:
             bs = potential_bs
 
-    message = f"TODO: A batch_size={bs} is recommended for training."
-
+    max_bs = 128
     if len(df) < 128:
         max_bs = 2
         for potential_bs in potential_batch_sizes:
             if potential_bs < len(df) * 0.9:
                 max_bs = potential_bs
-        additional_msg_for_small_df = f" Please have a maximum batch_size of {max_bs}."
-        message += additional_msg_for_small_df
-
-    return message
 
-
-def estimating_customization_job_time(df, recommend_hyperparameters_message):
-    recommended_batch_size = int(recommend_hyperparameters_message.split("=")[1].split()[0])
+    df_char_length = df.apply(lambda x: len(x.prompt) + len(x.completion), axis=1)
+    length_by_chars = sorted(list(df_char_length))
+    n_samples_under_99p5_limit = int(len(df_char_length) * 0.995 + 1)
+    max_char_length = length_by_chars[:n_samples_under_99p5_limit][-1]
```

Comment:
`length_by_chars[:n_samples_under_99p5_limit][-1]` -> `length_by_chars[n_samples_under_99p5_limit]`
won't this suffice?